File: app/src/main/java/com/micewine/emu/fragments/DeleteItemFragment.java
package com.micewine.emu.fragments;

import static com.micewine.emu.activities.MainActivity.selectedFilePath;
import static com.micewine.emu.activities.MainActivity.selectedFragmentId;
import static com.micewine.emu.adapters.AdapterGame.selectedGameName;
import static com.micewine.emu.adapters.AdapterPreset.clickedPresetName;
import static com.micewine.emu.adapters.AdapterPreset.clickedPresetType;
import static com.micewine.emu.fragments.Box64PresetManagerFragment.deleteBox64Preset;
import static com.micewine.emu.fragments.ControllerPresetManagerFragment.deleteControllerPreset;
import static com.micewine.emu.fragments.CreatePresetFragment.BOX64_PRESET;
import static com.micewine.emu.fragments.CreatePresetFragment.CONTROLLER_PRESET;
import static com.micewine.emu.fragments.CreatePresetFragment.VIRTUAL_CONTROLLER_PRESET;
import static com.micewine.emu.fragments.CreatePresetFragment.WINE_PREFIX_PRESET;
//import static com.micewine.emu.fragments.FileManagerFragment.deleteFile;
import static com.micewine.emu.fragments.FileManagerFragment.deleteFile;
import static com.micewine.emu.fragments.ShortcutsFragment.removeGameFromList;
import static com.micewine.emu.fragments.VirtualControllerPresetManagerFragment.deleteVirtualControllerPreset;
import static com.micewine.emu.fragments.WinePrefixManagerFragment.deleteWinePrefix;

import android.app.Dialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import com.micewine.emu.R;

public class DeleteItemFragment extends DialogFragment {
    private final int deleteType;

    public DeleteItemFragment(int deleteType) {
        this.deleteType = deleteType;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        LayoutInflater inflater = getLayoutInflater();
        View view = inflater.inflate(R.layout.fragment_delete_item, null);

        Button buttonContinue = view.findViewById(R.id.buttonContinue);
        Button buttonCancel = view.findViewById(R.id.buttonCancel);

        buttonContinue.setOnClickListener((v) -> {
            switch (deleteType) {
                case DELETE_GAME_ITEM: {
                    if (selectedFragmentId == 0) {
                        removeGameFromList(selectedGameName);
                    } else if (selectedFragmentId == 2) {
                        deleteFile(selectedFilePath);
                    }
                    break;
                }
                case DELETE_PRESET: {
                    switch (clickedPresetType) {
                        case CONTROLLER_PRESET: {
                            deleteControllerPreset(clickedPresetName);
                            break;
                        }
                        case VIRTUAL_CONTROLLER_PRESET: {
                            deleteVirtualControllerPreset(clickedPresetName);
                            break;
                        }
                        case BOX64_PRESET: {
                            if (!deleteBox64Preset(clickedPresetName)) {
                                Toast.makeText(requireContext(), R.string.remove_last_preset_error, Toast.LENGTH_SHORT).show();
                            }
                            break;
                        }
                        case WINE_PREFIX_PRESET: {
                            if (!deleteWinePrefix(clickedPresetName)) {
                                Toast.makeText(requireContext(), R.string.remove_last_wine_prefix_error, Toast.LENGTH_SHORT).show();
                            }
                            break;
                        }
                    }
                }
            }
            dismiss();
        });

        buttonCancel.setOnClickListener((v) -> dismiss());

        return new AlertDialog.Builder(requireActivity(), R.style.CustomAlertDialog).setView(view).create();
    }

    public final static int DELETE_GAME_ITEM = 0;
    public final static int DELETE_PRESET = 1;
}

File: app/src/main/java/com/micewine/emu/fragments/DeleteItemFragment.kt
package com.micewine.emu.fragments

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.widget.Button
import androidx.fragment.app.DialogFragment
import com.micewine.emu.R
import com.micewine.emu.activities.MainActivity.Companion.selectedFile
import com.micewine.emu.activities.MainActivity.Companion.selectedFragmentId
import com.micewine.emu.adapters.AdapterGame.Companion.selectedGameName
import com.micewine.emu.adapters.AdapterPreset.Companion.clickedPresetName
import com.micewine.emu.adapters.AdapterPreset.Companion.clickedPresetType
import com.micewine.emu.fragments.Box64PresetManagerFragment.Companion.deleteBox64Preset
import com.micewine.emu.fragments.ControllerPresetManagerFragment.Companion.deleteControllerPreset
import com.micewine.emu.fragments.CreatePresetFragment.Companion.BOX64_PRESET
import com.micewine.emu.fragments.CreatePresetFragment.Companion.CONTROLLER_PRESET
import com.micewine.emu.fragments.CreatePresetFragment.Companion.VIRTUAL_CONTROLLER_PRESET
import com.micewine.emu.fragments.CreatePresetFragment.Companion.WINEPREFIX_PRESET
import com.micewine.emu.fragments.FileManagerFragment.Companion.deleteFile
import com.micewine.emu.fragments.ShortcutsFragment.Companion.removeGameFromList
import com.micewine.emu.fragments.VirtualControllerPresetManagerFragment.Companion.deleteVirtualControllerPreset
import com.micewine.emu.fragments.WinePrefixManagerFragment.Companion.deleteWinePrefix

class DeleteItemFragment(private val deleteType: Int) : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val inflater = requireActivity().layoutInflater
        val view = inflater.inflate(R.layout.fragment_delete_item, null)

        val buttonContinue = view.findViewById<Button>(R.id.buttonContinue)
        val buttonCancel = view.findViewById<Button>(R.id.buttonCancel)

        val dialog = AlertDialog.Builder(requireContext(), R.style.CustomAlertDialog).setView(view).create()

        buttonContinue.setOnClickListener {
            when (deleteType) {
                DELETE_GAME_ITEM -> {
                    if (selectedFragmentId == 0) {
                        removeGameFromList(selectedGameName)
                    } else if (selectedFragmentId == 2) {
                        deleteFile(selectedFile)
                    }
                }
                DELETE_PRESET -> {
                    when (clickedPresetType) {
                        CONTROLLER_PRESET -> deleteControllerPreset(clickedPresetName)
                        VIRTUAL_CONTROLLER_PRESET -> deleteVirtualControllerPreset(clickedPresetName)
                        BOX64_PRESET -> deleteBox64Preset(clickedPresetName)
                        WINEPREFIX_PRESET -> deleteWinePrefix(clickedPresetName)
                    }
                }
            }

            dismiss()
        }

        buttonCancel.setOnClickListener {
            dismiss()
        }

        return dialog
    }

    companion object {
        const val DELETE_GAME_ITEM = 0
        const val DELETE_PRESET = 1
    }
}